hooks/useENS.ts

```diff
@@ -0,0 +1,31 @@
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+
+const useENS = (address?: string): { ensName: string | null; ensAvatar: string | null } => {
+	const [ensName, setENSName] = useState<string | null>(null);
+	const [ensAvatar, setENSAvatar] = useState<string | null>(null);
+
+	useEffect(() => {
+		let mounted = true;
+
+		(async () => {
+			if (address && ethers.utils.isAddress(address)) {
+				const provider = ethers.providers.getDefaultProvider();
```

Comment:
This will create a new provider for every single call to the function, and does not inherit the provider that already exists. I don't think this solution is viable at the current size of our leaderboard, which will continue to grow.

You should get the signer using the method, only once: [ENS for wallet](https://github.com/Kwenta/kwenta/blob/a2f09c957d38c7fa9fec0d9643c963f1b79563e2/sections/shared/Layout/AppLayout/Header/WalletActions.tsx#L37)